### PR TITLE
Remove the "dune compute" command

### PR DESCRIPTION
**Pull request by ghost**
The original intent of this command was to provide easy access to memoized functions from the command line. However, this feature is largely unused. Only 3 functions were public and they couldn't be called because of a parsing bug.

Getting rid of it allows to simplify a bit more the `Memo` API.

### CODE REVIEW

**Comment by jeremiedimino** [comment]
We had one use of `dune compute latest-lang-version` in the doc/ directory, which I replaced by `dune internal latest-lang-version`.

File: CHANGES.md [hunk 1]
```diff
@@ -131,6 +131,9 @@ Unreleased
   way to rerun all actions failed due to errors without restarting the build,
   e.g. via a Dune RPC call. (#4522, Andrey Mokhov)
 
+- Remove `dune compute`. It was broken and unused (#....,
```
Comment (aalekseyev) [inline CHANGES.md:134]: The PR number still needs to be filled in.

**Review by aalekseyev — APPROVED** [review]
Looks good, other than that changelog imperfection.

File: CHANGES.md (hunk 1 above)
Comment (jeremiedimino) [reply CHANGES.md:134]:
```suggestion
- Remove `dune compute`. It was broken and unused (#4540,
```

**Pull request merged**